app/src/main/java/com/google/droidjump/MainActivity.java

```diff
@@ -96,6 +104,37 @@ public PlayersClient getPlayersClient() {
         return playersClient;
     }
 
+    public void countTimeOfPlaying() {
+        String leaderboardId = getResources().getString(R.string.leaderboard_best_time);
+        if (savedSignedInAccount != null && LevelManager.getLastLevelIndex() != LevelManager.getLevelsLastIndex()) {
+            leaderboardsClient.loadCurrentPlayerLeaderboardScore(
+                    leaderboardId,
+                    LeaderboardVariant.TIME_SPAN_ALL_TIME,
+                    LeaderboardVariant.COLLECTION_PUBLIC).addOnSuccessListener(
+                    result -> {
+                        Timer timer = new Timer();
+                        TimerTask task = new TimerTask() {
+                            @Override
+                            public void run() {
+                                if (savedSignedInAccount != null) {
+                                    long time = ScoreManager.getScore(leaderboardId);
+                                    if (LevelManager.getLastLevelIndex() == LevelManager.getLevelsLastIndex()) {
+                                        ScoreManager.submitScore(leaderboardId, time);
+                                        timer.cancel();
+                                    }
+                                    time += TEN_SECONDS_IN_MILLISECONDS;
```

Comment:
is this the same as score where which obstacles is 10 points? Why is this the same for time?

Reply:
no, we have a best time leaderboard. I created a timer and every 10 seconds it updates SharedPreferences. If the user passes all levels, it will submit the score to leaderboard.